.github/workflows/mergify-ready.yml

```
@@ -107,9 +107,17 @@ jobs:
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
-
+      - name: Check if PR is up to date with master
+        id: is_based_on_main
+        run: |
+          result=$(git merge-base --is-ancestor $BASE_SHA $HEAD_SHA && echo true || echo false)
+          if [[ "$result" == "false" ]]; then
+            echo "PR is not up to date with master branch, skipping fixup commit check"
+          fi
+          echo "BRANCH_UP_TO_DATE=$result" >> $GITHUB_ENV
       - name: Check for fixup commits
         id: fixup-commits
+        if: ${{ env.BRANCH_UP_TO_DATE == 'true' }}
         run: |
           if [[ $(git rev-list "$BASE_SHA".."$HEAD_SHA" --grep="^\(fixup\|amend\|squash\)! " | wc -l) -eq 0 ]]; then
```

I'm wondering if it's not just simpler to avoid the extra step and do:
```suggestion
          if ! git merge-base --is-ancestor $BASE_SHA $HEAD_SHA; then
            echo "PR is not up to date with target branch, skipping fixup commit check"
          elif [[ $(git rev-list "$BASE_SHA".."$HEAD_SHA" --grep="^\(fixup\|amend\|squash\)! " | wc -l) -eq 0 ]]; then
```

---

i think this is the best idea to go with. tested it out here https://github.com/Agoric/agoric-sdk/actions/runs/11038333190/job/30661281557